### DISCUSSION TITLE
How to use getCellView(cell) for Minmap in V2

### DISCUSSION THREAD

**Discussion by tomsoftware** [antvis/X6#3081] [Q&A] [answered]
How to set a different rendering mode for nodes in Minimap in Version 2.

In V1 we can use
https://x6.antv.vision/en/docs/api/graph/minimap

```
...
  minimap: {
    enabled: true,
    container: minimapContainer,
    graphOptions: {
      async: true,
      getCellView(cell) {
         ...
      },
      createCellView(cell) {
       ....
      },
```

In V2
https://x6.antv.antgroup.com/tutorial/plugins/minimap
I do not know how to do so?

**Comment by NewByVector** [comment, marked as answer]
You can use `createCellView` to customize the element view, which returns a `CellView` that will replace the default view, if it returns `null` it will not render, if it returns `undefined` it will render as default.

**Reply by NewByVector** [reply]
createCellView support on @antv/x6@2.1.2

**Comment by tomsoftware** [comment]
Perfect! Thank you!

I finally achieve the V1 behaviour in version V2 by using:

```
this.graph.use(
            new MiniMap({
                container: destContainer,
                width: 400,
                height: 120,
                graphOptions: {
                    async: true,
                    createCellView: (cell) => {
                        if (cell.isEdge()) {
                            return null;
                        }
                        if (cell.isNode()) {
                            if (this.graph) {
                                return new SimpleNodeView(cell, { graph: this.graph });
                            }
                        }
                    },
                },
            },)
        );

```

Can you check why I need to inject the graph itself as an option? If I don't, I get a runtime error:

```
TypeError: Cannot read properties of undefined (reading 'options')
    at SimpleNodeView.can (cell.js:207:1)
```

 SimpleNodeView.can (cell.js:207:1) : 
```
 can(feature) {
        let interacting = this.graph.options.interacting; <--- error
```

Or may I'm doing some other mistakes?

**Reply by NewByVector** [reply]
Do not use the arrow function with createCellView.

**Reply by NewByVector** [reply]
我们在 2.9.3 版本后，`createCellView` 只要传入 SimpleNodeView，而不需要去 `new SimpleNodeView(cell, { graph })`
